app/controllers/spree/subscriptions_controller.rb

```diff
@@ -23,25 +23,22 @@ def update
     end
 
     def cancel
+      return respond_with_error(t(".stripe_subscription_id_not_found"), 422) unless @subscription.stripe_subscription_id.present?
+    
+      unless @subscription.cancel
+        return respond_with_error(t(".error"), 422)
+      end
+    
+      cancel_stripe_subscription
```

Comment:
i think you didn't get my point. We need to check stripe api for failure, rather you add a check on your rails model but just moved stripe api into a seperate method.